sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -87,12 +87,16 @@ public static int calculateBitSetWidthInBytes(int numFields) {
 
   public static boolean isFixedLength(DataType dt) {
     if (dt instanceof DecimalType) {
-      return ((DecimalType) dt).precision() < Decimal.MAX_LONG_DIGITS();
+      return ((DecimalType) dt).precision() <= Decimal.MAX_LONG_DIGITS();
     } else {
       return settableFieldTypes.contains(dt);
     }
   }
 
+  public static boolean isSettable(DataType dt) {
```

Comment:
ah ic you can't combine them since it's used in other places as well -- in that case renaming this to isMutable makes more sense.